routes/utils.js

```diff
@@ -430,3 +433,72 @@ function _freeId (objType, id) {
         }
     });
 }
+
+
+
+// generic function for send notif (and email)
+
+function get_mail_config () {
+    var MAIL_CONFIG = {};
+    // todo: more and more ugly init...
+    if (CONFIG[MAILER]) { MAIL_CONFIG = CONFIG[MAILER]; }
+    // todo: find a cleaner way to allow registration if no mail are configured
+    if (!MAIL_CONFIG.origin) { MAIL_CONFIG.origin = 'nomail@nomail.org'; }
+    return MAIL_CONFIG;
+}
+
+async function gen_mail_opt (options, variables)
+{
+    var MAIL_CONFIG = get_mail_config();
+    // todo: check if each option exist and use default value
+    let name = options['name'];
+    let destinations = options['destinations'];
+    let subject = GENERAL_CONFIG.name + ' ' + options['subject'];
+
+    //find message
+    let message = CONFIG.message[name].join('\n');
+    let html_message = CONFIG.message[name + '_html'].join('');
```

Comment:
should we not manage case where name:
1) is not defined in config
2) we have *name* (text) but not *name*_html (html version). In this case use text only.

Reply:
Yes I have add a comment with a  7 line before, so it is in my "todo list" ;)

Reply:
https://github.com/genouest/genouestaccountmanager/pull/166/commits/85ec1bdc1ec57c5022281b1b1304d9c89ebd0b98